### PR TITLE
Add cli commands and rename roku-deploy functions, reorganize functions

### LINKED ISSUE #70: CLI commands for various roku-deploy actions
We should be able to do everything in roku-deploy from the command-line. Things like move-to-stage, create zip from a folder, etc. 

Here's a good example of a cli file that this feature can be modelled after. 

https://github.com/rokucommunity/ropm/blob/master/src/cli.ts

We want to expose most of the public functions on the [RokuDeploy](https://github.com/rokucommunity/roku-deploy/blob/master/src/RokuDeploy.ts) class. 

So tasks for this:
 - install yargs and add distinct cli commands for all the important RokuDeploy functions, with their args
 - make dedicated commands/ folder with a file per command (like in ropm https://github.com/rokucommunity/ropm/tree/master/src/commands)

Here are a few examples of how we'd use it (args and names might need to be changed, but this gets the point across):
```bash
npx roku-deploy copy-to-staging --stagingDir ./staging
npx roku-deploy create-zip --stagingDir ./staging --file app.zip
npx roku-deploy deploy --host 192.168.1.123 --password --file app.zip
```

### LINKED ISSUE #138: Remove rokudeploy.json and bsconfig.json loading from rokudeploy functions.
Once #70 lands, we need to move the json file loading OUT of the inner rokuDeploy functions and into the cli instead. The CLI should load the json files in the same way, and augment the options object with the data found in the json files (cli args override json obj props).

### PR DESCRIPTION
**Pull request by TwitchBronBron**
Adds all method and CLI renames we want to do. This thing is totally broken though, needs lots of love. :D 

Fixes #70
Fixes #138

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -101,7 +101,7 @@
         ],
         "sourceMap": true,
         "instrument": true,
-        "check-coverage": true,
+        "check-coverage": false,
```
Comment (TwitchBronBron) [inline package.json:104]: We'll re-enable this once most of v4 is finished. No need to chase 100% right now

File: src/RokuDeploy.spec.ts [hunk 2]
```diff
@@ -699,182 +637,6 @@ describe('index', () => {
 
     });
 
-    describe('createPackage', () => {
```
Comment (TwitchBronBron) [inline src/RokuDeploy.spec.ts:702]: I think we should bring back most of these tests. Looks like maybe we just need to tweak some of them to test against the new rokuDeploy function names?

File: src/RokuDeploy.spec.ts [hunk 3]
```diff
@@ -2169,96 +1938,30 @@ describe('index', () => {
...
-            expect(() => rokuDeploy['normalizeFilesArray'](<any>[new Date()])).to.throw();
-            expect(() => rokuDeploy['normalizeFilesArray'](<any>[1])).to.throw();
-            expect(() => rokuDeploy['normalizeFilesArray'](<any>[{ src: true }])).to.throw();
-            expect(() => rokuDeploy['normalizeFilesArray'](<any>[{ src: /asdf/ }])).to.throw();
-            expect(() => rokuDeploy['normalizeFilesArray'](<any>[{ src: new Date() }])).to.throw();
-            expect(() => rokuDeploy['normalizeFilesArray'](<any>[{ src: 1 }])).to.throw();
-        });
-    });
-
-    describe('deploy', () => {
```
Comment (TwitchBronBron) [inline src/RokuDeploy.spec.ts:2183]: Let's try and re-enable these tests...might need to move them to test the `ExecCommand` instead if that's where the full flow happens?

File: src/RokuDeploy.spec.ts [hunk 4]
```diff
@@ -3291,231 +2933,41 @@ describe('index', () => {
...
-        });
-
-        it('handles non-falsey values', () => {
-            expect(rokuDeploy.normalizeRootDir(cwd)).to.equal(cwd);
-            expect(rokuDeploy.normalizeRootDir('./')).to.equal(cwd);
-            expect(rokuDeploy.normalizeRootDir('./testProject')).to.equal(path.join(cwd, 'testProject'));
-        });
-    });
-
-    describe('retrieveSignedPackage', () => {
```
Comment (TwitchBronBron) [inline src/RokuDeploy.spec.ts:3417]: I think these tests need to be restored as well

File: src/RokuDeploy.spec.ts [hunk 5]
```diff
@@ -3748,85 +3090,25 @@ describe('index', () => {
...
             const writeStream = await writeStreamPromise;
             writeStream.write('test');
             writeStream.close();
 
             await finalPromise;
-            expect(getToFileIsResolved).to.be.true;
-        });
-    });
-
-    describe('deployAndSignPackage', () => {
```
Comment (TwitchBronBron) [inline src/RokuDeploy.spec.ts:3773]: Restore these tests. Might need to test the ExecCommand to test these flows

File: src/RokuDeploy.ts [hunk 6]
```diff
@@ -581,6 +422,11 @@ export class RokuDeploy {
         let parsedManifest = await this.parseManifest(manifestPath);
         let appName = parsedManifest.title + '/' + parsedManifest.major_version + '.' + parsedManifest.minor_version;
 
+        //prevent devId mismatch (if devId is specified)
+        if (options.devId && options.devId !== await this.getDevId()) {
+            throw new Error('devId mismatch. nope, not gonna sign');
```
Comment (TwitchBronBron) [inline src/RokuDeploy.ts:427]: Let's make a better error message here. Maybe something like:
```
`Package signing cancelled: devId '${options.devId}' does not match on-device devId `${onDeviceDevId}`
```

File: src/RokuDeploy.ts [hunk 7]
```diff
@@ -1059,21 +824,25 @@ export class RokuDeploy {
         }
     }
 
+    /**
+     * @param options
```
Comment (TwitchBronBron) [inline src/RokuDeploy.ts:828]: Need a better description here

File: src/cli.spec.ts [hunk 8]
```diff
@@ -163,58 +160,12 @@ describe('cli', () => {
         });
     });
 
-    it('Retrieves a signed package', async () => {
```
Comment (TwitchBronBron) [inline src/cli.spec.ts:166]: Can we restore these tests and make them work with how the CLI runs? Or move them to a different file and test the ExecCommand instead?

File: src/cli.ts [hunk 9]
```diff
@@ -1,160 +1,243 @@
...
     }, (args: any) => {
-        return new PublishCommand().run(args);
+        return new ExecCommand(
+            'close|rekey|stage|zip|delete|close|sideload|squash|sign',
+            args
+        ).run();
+    })
+
+    .command('exec', 'larger command for handling a series of smaller commands', () => {
+        return attachCommonArgs;
```
Comment (TwitchBronBron) [inline src/cli.ts:83]: Let's just go ahead and inline the options here.

Comment (TwitchBronBron) [inline src/cli.ts:83]: I think this command needs a little more description. Maybe even a few examples using .example() (or whatever yargs provides for showing examples)

File: src/device.spec.ts [hunk 10]
```diff
@@ -62,31 +61,5 @@ describe('device', function device() {
...
-    describe('deployAndSignPackage', () => {
-        it('works', async () => {
-            await rokuDeploy.deleteInstalledChannel(options as any);
-            await rokuDeploy.rekeyDevice(options as any);
-            expectPathExists(
-                await rokuDeploy.deployAndSignPackage(options as any)
-            );
-        });
-    });
+    console.log(options); // So there are no errors about unused variable
```
Comment (TwitchBronBron) [inline src/device.spec.ts:64]: If we're not actually using options, let's delete it. Either way, delete this comment.

**Review by TwitchBronBron — COMMENTED** [review]
Left a few comments. Looking great!

File: src/RokuDeploy.spec.ts (hunk 3 above)
Comment (MilapNaik) [reply src/RokuDeploy.spec.ts:2183]: Done

File: src/cli.spec.ts (hunk 8 above)
Comment (MilapNaik) [reply src/cli.spec.ts:166]: Decided not to reimplement

File: src/RokuDeploy.ts (hunk 6 above)
Comment (MilapNaik) [reply src/RokuDeploy.ts]: Done and added test for it

File: src/device.spec.ts (hunk 10 above)
Comment (MilapNaik) [reply src/device.spec.ts]: done

File: src/RokuDeploy.ts (hunk 7 above)
Comment (MilapNaik) [reply src/RokuDeploy.ts:832]: Done

File: src/cli.ts (hunk 9 above)
Comment (MilapNaik) [reply src/cli.ts]: Done

Comment (MilapNaik) [reply src/cli.ts]: Done

File: src/RokuDeploy.spec.ts (hunk 2 above)
Comment (MilapNaik) [reply src/RokuDeploy.spec.ts:702]: Done

File: src/RokuDeploy.spec.ts (hunk 5 above)
Comment (MilapNaik) [reply src/RokuDeploy.spec.ts:3773]: Done

File: src/RokuDeploy.spec.ts (hunk 4 above)
Comment (MilapNaik) [reply src/RokuDeploy.spec.ts:3417]: Done

File: src/RokuDeploy.ts [hunk 11]
```diff
@@ -1284,6 +994,7 @@ export interface PublishOptions {
     retainDeploymentArchive?: boolean;
```
Comment (TwitchBronBron) [inline src/RokuDeploy.ts:994]: Missing `outFile?: string;`

File: src/RokuDeploy.ts [hunk 12]
```diff
@@ -1253,29 +939,53 @@ export interface GetDeviceInfoOptions {
...
+
+export interface SendTextOptions extends SendKeyEventOptions {
+    action?: 'keypress';
+    text: string;
 }
 
-export interface CreatePackageOptions {
+export interface CloseChannelOptions {
+    host: string;
+    remotePort: number;
```
Comment (TwitchBronBron) [inline src/RokuDeploy.ts:973]:
```suggestion
    remotePort?: number;
```

File: src/cli.ts [hunk 13]
```diff
@@ -1,160 +1,242 @@
...
+import { KeyDownCommand } from './commands/KeyDownCommand';
 
 void yargs
-    .command('prepublishToStaging', 'Copies all of the referenced files to the staging folder', (builder) => {
+
+    .command('bundle', 'execute build actions for bundling app', (builder) => {
         return builder
-            .option('stagingDir', { type: 'string', description: 'The selected staging folder', demandOption: false })
-            .option('rootDir', { type: 'string', description: 'The selected root folder to be copied', demandOption: false });
+            .option('rootDir', { type: 'string', description: 'The selected root folder to be copied', demandOption: false })
```
Comment (TwitchBronBron) [inline src/cli.ts:24]: Add missing `outFile` option

**Pull request merged**